test/jdk/java/lang/management/ThreadMXBean/VirtualThreadDeadlocks.java

```diff
@@ -61,6 +64,8 @@ public class VirtualThreadDeadlocks {
      * VV = test deadlock with two virtual threads
      */
     public static void main(String[] args) throws Exception {
+        // need at least two carrier threads due to pinning
+        VThreadRunner.ensureParallelism(2);
```

Comment:
This test doesn't need to restore it so it doesn't keep a reference and not an issue if it is GC'ed.

Reply:
It's not a matter of whether or not the test needs to restore it. It _will_ restore it if there is a GC, and if this happens before the test completes, it could find itself without enough carrier threads.

Reply:
No, restoration requires a call to close. Maybe you are assuming that AutoCloseable implementations setup a cleaner to do that?

Reply:
Yes, that was my point of confusion. I thought collection triggered calling close, but I see now when you want it to be closed, you rely on try-with-resources to call close() method.